lib/membrane_http_adaptive_stream/bandwidth_calculator.ex

```diff
@@ -13,53 +15,21 @@ defmodule Membrane.HTTPAdaptiveStream.BandwidthCalculator do
   # and risk that client will use track with actual bandwidth that is beyond its capabilities.
   @default_bandwidth 2_560_000
 
-  @spec calculate_bandwidth(Track.t()) :: integer()
-  def calculate_bandwidth(track) do
-    target_duration = track.target_segment_duration / Time.second()
-    segments_sequences = generate_subsequences(track.segments |> Enum.to_list())
-
-    segments_meta =
-      segments_sequences
-      |> Enum.map(
-        &{
-          Enum.map(&1, fn sg -> 8 * sg.byte_size end) |> Enum.sum(),
-          Enum.map(&1, fn sg -> sg.duration / Time.second() end)
-          |> Enum.reduce(fn acc, x -> acc + x end)
-        }
-      )
-      |> Enum.filter(fn {_bits_size, duration} -> duration != 0.0 end)
-
-    segments_bitrates =
-      segments_meta |> Enum.map(fn {bits_size, duration} -> bits_size / duration end)
-
-    # According to HLS rfc only segment subsequences with duration between 0.5 and 1.5 of target duration should be
-    # considered
-    validated_segments_meta =
-      segments_meta
-      |> Enum.filter(fn {_bits_size, duration} ->
-        duration >= 0.5 * target_duration and duration <= 1.5 * target_duration
-      end)
-
-    validated_segments_bitrates =
-      validated_segments_meta |> Enum.map(fn {bits_size, duration} -> bits_size / duration end)
-
-    cond do
-      Enum.empty?(segments_meta) ->
-        @default_bandwidth
-
-      Enum.empty?(validated_segments_meta) ->
-        segments_bitrates |> Enum.max(&Ratio.>=/2) |> Ratio.floor()
-
-      true ->
-        validated_segments_bitrates |> Enum.max(&Ratio.>=/2) |> Ratio.floor()
+  @spec calculate_bandwidth(Track.t(), integer()) :: integer()
+  def calculate_bandwidth(track, segments_number \\ 20) do
+    segments =
+      track.segments
+      |> Enum.to_list()
+      |> Enum.take(-segments_number)
+      |> Enum.filter(&(&1.duration > 0))
```

Comment:
are there any segments with duration 0?

Reply:
Apparently there are, perhaps when a new segment is created it has duration of 0, which later changes to a meaningful value. It is just an assumption tho.

Reply:
It seems it's indeed the case when we have partial segments. But in that case, it would be better to check for `complete?`